src/components/UrlRewritePreview.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies.
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * UrlRewritePreview component for GatherPress.
+ *
+ * This component renders a preview of the rewritten url based on the specified string.
+ * It listens for the 'input' event on the input field with the specified name and updates
+ * the state with the new rewritten url.
+ *
+ * @since 1.0.0
+ *
+ * @param {Object} props             - Component props.
+ * @param {Object} props.attrs       - Component attributes.
+ * @param {string} props.attrs.name  - The name of the input field.
+ * @param {string} props.attrs.value - The initial value of the input field (rewritten url).
+ *
+ * @return {JSX.Element} The rendered React component.
+ */
+const UrlRewritePreview = (props) => {
+	const { name, value, suffix } = props.attrs;
+	const [rewrittenUrlPart, setRewrittenUrlPart] = useState(value);
+
+	const input = document.querySelector(`[name="${name}"]`);
+
+	input.addEventListener(
+		'input',
+		(e) => {
+			setRewrittenUrlPart(e.target.value);
+		},
+		{ once: true }
+	);
+
+	return (
+		<>
+			{window.GatherPress.urls.homeUrl + '/'}
```

Comment:
Sorry, missed usage of this. There's a `getFromGlobal` helper you can use here. Has build in error handling and such.
`getFromGlobal('urls.homeUrl')`

Reply:
Nice! Also, maybe we set the `/` where we set the `homeUrl` so we don't have to append it here. `get_home_url( '/' ); `